type.go

```diff
@@ -804,6 +828,17 @@ func (t byteArrayType) AssignValue(dst reflect.Value, src Value) error {
 		dst.SetString(string(v))
 	case reflect.Slice:
 		dst.SetBytes(copyBytes(v))
+	case reflect.Pointer:
+		underlying := reflect.TypeOf(dst).Elem().Kind()
+		switch underlying {
+		case reflect.String:
+			str := string(v)
+			dst.Set(reflect.ValueOf(&str))
+		default:
+			// trigger a panic for type correction
+			val := reflect.ValueOf(string(v))
+			dst.Set(val)
+		}
```

Comment:
Could we use the same recursive call to `AssignValue` here as well?

Reply:
Sure, made the procedure the same for all pointer handlings.